packages/amplify-category-function/src/provider-utils/awscloudformation/utils/updateDependentFunctionCfn.ts

```diff
@@ -0,0 +1,85 @@
+import { $TSContext, $TSObject, JSONUtilities } from 'amplify-cli-core';
+import { FunctionParameters } from 'amplify-function-plugin-interface';
+import { getResourcesForCfn, generateEnvVariablesForCfn } from '../service-walkthroughs/execPermissionsWalkthrough';
+import { updateCFNFileForResourcePermissions } from '../service-walkthroughs/lambda-walkthrough';
+import { loadFunctionParameters } from './loadFunctionParameters';
+import path from 'path';
+import { functionParametersFileName } from './constants';
+import { category } from '../../../constants';
+
+export async function updateDependentFunctionsCfn(
+  context: $TSContext,
+  dependentFunctionResource: $TSObject[],
+  backendDir: string,
+  modelsDeleted: string[],
+  apiResource: string,
+) {
+  // remove function parameters from if there is dependency
+  /*
+    1) get the List of functions
+    2) check the functions parameters.json for dependency and api
+    3) update CFN and functional parameters file if table is deleted.
+    */
+  // if the function is deleted -> not possible as have to remove api
+  // initialize function parameters for update
+  for (const lambda of dependentFunctionResource) {
+    const resourceDirPath = path.join(backendDir, category, lambda.resourceName);
+    const currentParameters = loadFunctionParameters(context, resourceDirPath);
+    const selectedCategories = currentParameters.permissions;
+    let categoryPolicies = [];
+    let permissions = {};
+    let resources = [];
+    let resourcePolicy = {};
+    const functionParameters: Partial<FunctionParameters> = {
+      resourceName: lambda.resourceName,
+      environmentMap: {
+        ENV: {
+          Ref: 'env',
+        },
+        REGION: {
+          Ref: 'AWS::Region',
+        },
+      },
+    };
+
+    for (const selectedCategory of Object.keys(selectedCategories)) {
+      // update function parameters resouce parameters with deleted models data
+      // remove the deleted @model
+      const selectedResources = selectedCategories[selectedCategory];
+      for (const resourceName of Object.keys(selectedResources)) {
+        if (!modelsDeleted.includes(resourceName)) {
```

Comment:
would it be easier to remove the policies of the resources that are in the deleted list rather than adding resources not in the list?

Reply:
yeah I tried that earlier when started this but the policies and variables for dependency are all over cfn template. I followed the same approach when we update the  functions with resource access permission. Refractored the ```execPermissionsWalkthrough.ts``` file into functions and used those component here.

Also added tests for refractored ```execPermissionsWalkthrough.ts```  file.